M2.el

```diff
@@ -306,11 +306,11 @@ can be executed with \\[M2-send-to-program]."
   interpreter using \\[comint-send-input]."
   (interactive)
   (cond ((save-excursion
-	   (search-backward-regexp "\\({\\*\\|^\\)")
-	   ;; example: {*FunctionBody[../../d/startup.m2.in:123:19-123:21]*}
-	   ;; example: {*Function[../../m2/res.m2:191:40-202:36]*}
+	   (search-backward-regexp "\\(-\\*\\|^\\)")
```

Comment:
Do you need to add `--` to this regexp? I think the idea is that if you press enter while in the middle of the line, the cursor should backtrack to this point, then check for one of the expressions. Adding `--` should allow something like this to work also:
```m2
blah blah -- ../file.m2
```

Reply:
This particular section of the code is just for the `-*Function[...]*-` and `-*FunctionBody[...]*-` stuff, which necessarily start with `-*`. I'm not sure why we even bothered with the `^` option...

However, your suggestion should work for all the other parts.  I'll play around with it a bit more.

Reply:
>  I'm not sure why we even bothered with the `^` option...

I quickly figured out why -- Emacs throws a "Search failed" error if we hit enter and don't match the regex!

Reply:
I played around with `search-backward-regexp` for the other cases for a bit, but ended up sticking with `beginning-of-line` and just tossing `.*` in front of the two comments.